incubator/hnc/internal/reconcilers/hierarchy_config.go

```diff
@@ -212,11 +218,36 @@ func (r *HierarchyConfigReconciler) syncWithForest(log logr.Logger, nsInst *core
 	r.syncLabel(log, nsInst, ns)
 }
 
+// syncExternalNamespace sets external tree labels to the namespace in the forest
+// if the namespace is an external namespace not managed by HNC.
+func (r *HierarchyConfigReconciler) syncExternalNamespace(log logr.Logger, nsInst *corev1.Namespace, ns *forest.Namespace) {
+	manager := nsInst.Annotations[api.AnnotationManagedBy]
+	if manager == "" || manager == api.MetaGroup {
+		return
```

Comment:
You should explicitly set ns.ExternalTreeLabels to `nil` here in case this _used_ to be an external namespace but isn't anymore (and add a unit test for this).

Reply:
Good catch! Yes, I should set `ns.ExternalTreeLabels` to `nil` here. Additionally, I added if the namespace is just converted from external to internal, enqueue all the descendants to update the tree labels (remove propagated external tree labels). I added the unit test first and caught the error of not updating the descendants.

After this change, the new test case is passed. PTAL at the test cases.

Reply:
Added "enqueue descendants" for the other case too (converted from internal to external) and the new test case.